### ISSUE TITLE
Problem with `release-docker` job

### ISSUE THREAD

**Issue by lukesteensen** [timberio/vector#2780]
See failed run here: https://app.circleci.com/pipelines/github/timberio/vector/8613/workflows/b7953ddb-e34c-486f-a04d-a3a0627ab992/jobs/95711

**Comment by Hoverbear** [comment]
So this script proxies to `scripts/build-docker.sh` which when I run results in:

https://github.com/timberio/vector/blob/3e4360a2db28b1837e49933e3a6a23b29f97b136/scripts/release-docker.sh#L5

```
vector on  nix-env-test [$?] is 📦 v0.10.0 via ⬢ v12.18.0 via 🦀 v1.43.1 via ❄️  impure took 2s 
❯ ./scripts/build-docker.sh
++ scripts/util/release-channel.sh
+ CHANNEL=latest
++ scripts/version.sh
+ VERSION=0.10.0
++ date -u +%Y-%m-%d
+ DATE=2020-06-11
+ PUSH=
+ PLATFORM=
+ REPO=timberio/vector
+ echo 'Building timberio/vector:* Docker images'
Building timberio/vector:* Docker images
+ [[ latest == \l\a\t\e\s\t ]]
+ VERSION_EXACT=0.10.0
++ echo 0.10.0
++ sed 's/\.[0-9]*$/.X/g'
+ VERSION_MINOR_X=0.10.X
++ echo 0.10.0
++ sed 's/\.[0-9]*\.[0-9]*$/.X/g'
+ VERSION_MAJOR_X=0.X
+ for VERSION_TAG in "$VERSION_EXACT" "$VERSION_MINOR_X" "$VERSION_MAJOR_X" latest
+ build alpine 0.10.0
+ local BASE=alpine
+ local VERSION=0.10.0
+ local TAG=timberio/vector:0.10.0-alpine
+ local DOCKERFILE=distribution/docker/alpine/Dockerfile
+ '[' -n '' ']'
+ docker build --tag timberio/vector:0.10.0-alpine target/artifacts -f distribution/docker/alpine/Dockerfile
Sending build context to Docker daemon   4.15MB
Step 1/9 : FROM alpine:3.10 AS builder
 ---> be4e4bea2c2e
Step 2/9 : WORKDIR /vector
 ---> Using cache
 ---> 54e176cb1050
Step 3/9 : COPY vector-*-unknown-linux-musl*.tar.gz ./
COPY failed: no source files were specified
```

The `Dockerfile` depends on the musl target: https://github.com/timberio/vector/blob/5fa10916882cd07ee6c6726be10227b321f5880c/distribution/docker/alpine/Dockerfile#L5

So I'm building that first with `make build-x86_64-unknown-linux-musl`...

**Comment by Hoverbear** [comment]
@lukesteensen is this reproducable? Can we rerun the job?

**Comment by lukesteensen** [comment]
We've rerun it a couple of time and it has yet to pass. Here are the three other examples I found:

https://app.circleci.com/pipelines/github/timberio/vector/8366/workflows/b21a2212-72c4-4a2e-94ae-4477999f007e/jobs/95688

https://app.circleci.com/pipelines/github/timberio/vector/8366/workflows/7f14a64c-843a-41cf-9a1e-3a13d2fab839/jobs/95675

https://app.circleci.com/pipelines/github/timberio/vector/8366/workflows/3df5fcbc-78b5-4c04-9e7f-f8328d871bb1/jobs/95671

**Comment by binarylogic** [comment]
This work probably overlaps with https://github.com/timberio/vector/issues/2797.

**Comment by Hoverbear** [comment]
Alright lets fix it then. :)

**Comment by Hoverbear** [comment]
I think this is resolved after #2797
